src/Microsoft.Azure.WebJobs.Extensions.Storage/DeadlockHandler.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Timers;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage
+{
+    internal static class DeadlockHandler
+    {
+        private static readonly TimeSpan DefaultDeadlockTimeout = TimeSpan.FromMinutes(2);
```

Comment:
How'd you arrive at this number? I get that we're just trying to pick a number higher than any reasonable expected duration.

Reply:
Just a reasonable choice. From my testing, successes are very fast (maybe a couple seconds max, even with 5000 blobs) -- but I have no idea what the outliers would be. I figured 2 minutes of paused polling is better than the infinity minutes we see today. Open to other suggestions.